pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -119,23 +120,29 @@ func withUserAgent(sess *session.Session) *session.Session {
 }
 
 // Create a node given the constraints.
-func (c *CloudProvider) Create(ctx context.Context, constraints *v1alpha4.Constraints, instanceTypes []cloudprovider.InstanceType, callback func(*v1.Node) error) chan error {
+func (c *CloudProvider) Create(ctx context.Context, constraints *v1alpha4.Constraints, instanceTypes []cloudprovider.InstanceType, quantity int, callback func(*v1.Node) error) chan error {
 	return c.creationQueue.Add(func() error {
-		return c.create(ctx, constraints, instanceTypes, callback)
+		return c.create(ctx, constraints, instanceTypes, quantity, callback)
 	})
 }
 
-func (c *CloudProvider) create(ctx context.Context, constraints *v1alpha4.Constraints, instanceTypes []cloudprovider.InstanceType, callback func(*v1.Node) error) error {
+func (c *CloudProvider) create(ctx context.Context, constraints *v1alpha4.Constraints, instanceTypes []cloudprovider.InstanceType, quantity int, callback func(*v1.Node) error) error {
 	vendorConstraints, err := v1alpha1.NewConstraints(constraints)
 	if err != nil {
 		return err
 	}
-	// 3. Create instance
-	node, err := c.instanceProvider.Create(ctx, vendorConstraints, instanceTypes)
+	// Create instance
+	nodes, err := c.instanceProvider.Create(ctx, vendorConstraints, instanceTypes, quantity)
 	if err != nil {
-		return fmt.Errorf("launching instance, %w", err)
+		return fmt.Errorf("launching %d instance(s), %w", quantity, err)
```

Comment:
can't you now have the case where `nodes` has some nodes in it _and_ you have an `err` that isn't `nil`? (if so I don't think you want to just return here without doing something with the nodes you did get?)